client/src/popups/ShareMyContentModal.tsx

```diff
@@ -0,0 +1,317 @@
+          sharedWith={sharedWith}
+          parentSharedWith={parentSharedWith}
+        />
+      )}
+
+      <FormControl isInvalid={addEmailError ? true : false} marginTop="20px">
+        <FormLabel>Add people</FormLabel>
+        <HStack>
+          <Input
+            type="email"
+            name="email"
+            placeholder="Email address"
+            value={emailInput}
+            data-test="Email address"
+            onChange={(e) => {
+              if (e.target.value !== emailInput) {
+                setInputHasChanged(true);
+                setEmailInput(e.target.value);
+              }
+            }}
+            width="90%"
+            onBlur={() => {
+              if (inputHasChanged) {
+                addEmail();
+              }
+            }}
+            onKeyDown={(e) => {
+              if (e.key == "Enter" && inputHasChanged) {
+                addEmail();
+              }
+            }}
+          />
+
+          <SpinnerWhileFetching state={addEmailFetcher.state} />
+        </HStack>
+
+        {addEmailError && <FormErrorMessage>{addEmailError}</FormErrorMessage>}
+      </FormControl>
+    </>
+  );
+}
+
+function SharePublicly({
+  isPublic,
+  parentIsPublic,
+  contentType,
+  settings,
+  closeModal,
+}: {
+  isPublic: boolean;
+  parentIsPublic: boolean;
+  contentType: ContentType;
+  settings: Awaited<ReturnType<typeof settingsLoader>>;
+  closeModal: () => void;
+}) {
+  const fetcher = useFetcher();
+
+  const { allLicenses } = useOutletContext<EditorContext>();
+
+  if (parentIsPublic) {
+    return <p>Parent is public.</p>;
+  } else if (isPublic) {
+    return (
+      <>
+        <p>Content is public.</p>
+        <Button
+          mt="1rem"
+          size="sm"
+          colorScheme="blue"
+          onClick={() => {
+            fetcher.submit(
+              {
+                path: "share/setContentIsPublic",
+                isPublic: false,
+              },
+              { method: "post", encType: "application/json" },
+            );
+            closeModal();
+          }}
+        >
+          Unshare
+        </Button>
+      </>
+    );
+  } else {
+    const disableSubmit = settings.contentFeatures.length === 0;
+
+    return (
+      <Box>
+        <Heading size="md">Review/update this activity&apos;s settings</Heading>
+
+        <FormControl isRequired mt="2rem">
+          <FormLabel fontSize="xl">Categories</FormLabel>
+          <Box ml="1rem">
+            <EditContentFeatures
+              contentFeatures={settings.contentFeatures}
+              allContentFeatures={settings.allContentFeatures}
+            />
+          </Box>
+        </FormControl>
+
+        <FormControl mt="2rem">
+          <FormLabel fontSize="xl">Classifications</FormLabel>
+          <Box ml="1rem">
+            <EditClassifications classifications={settings.classifications} />
+          </Box>
+        </FormControl>
+
+        <FormControl isRequired mt="2rem">
```

Comment:
Omit `isRequired` here, as there isn't a way to not have this setting specified.